src/modules/aws_ec2/rpcs/invoke.ts

```diff
@@ -0,0 +1,63 @@
+import _ from 'lodash';
+
+import { EC2 } from '@aws-sdk/client-ec2';
+
+import { AwsEc2Module } from '..';
+import { crudBuilder2 } from '../../../services/aws_macros';
+import {
+  Context,
+  PostTransactionCheck,
+  PreTransactionCheck,
+  RpcBase,
+  RpcResponseObject,
+} from '../../interfaces';
+
+/**
+ * Method for invoking an AWS EC2 SDK function
+ */
+export class InvokeRpc extends RpcBase {
+  preTransactionCheck = PreTransactionCheck.NO_CHECK;
+  postTransactionCheck = PostTransactionCheck.NO_CHECK;
```

Comment:
Since this method could invoke whatever function and that could imply changes in the cloud I suggest running this without a transaction check, this would make it try to call `iasql_begin` before starting and then call `iasql_commit` when it's done:
```suggestion
```

Reply:
Yes I'm going to do that. I just added those two for testing purposes to not slow down the testing process.

Reply:
BTW now that I'm thinking, I think we should keep it as-is. The user can sync on demand, but having this would slow down the process for them and since this is a pro feature, we can communicate that they should sync the state periodically.

Reply:
hmm, I understand it will delay a little bit the process but I think it could be risky because this lead to a case where the user can do 2 operations at the same time, while this rpc could be creating something in the cloud another `apply` could be running and deleting what has just been created. 

The right  solution actually is to use: 
- `preTransactionCheck = PreTransactionCheck.WAIT_FOR_LOCK;`
- `postTransactionCheck = PostTransactionCheck.UNLOCK_ALWAYS;`

With these, it will be safe to run in a transaction but it will not add the delay for the `commit`